### PR TITLE
`Paywalls`: changed images representation to an object

### PR DESCRIPTION
**Pull request by NachoSoto**
This is much better, no longer relying on array index.

### CODE REVIEW

File: RevenueCatUI/Data/TemplateViewConfiguration+Images.swift [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+//  Created by Nacho Soto on 7/25/23.
+//
+
+import Foundation
+import RevenueCat
+
+@available(iOS 16.0, macOS 13.0, tvOS 16.0, *)
+extension TemplateViewConfiguration {
+
+    var headerImageURL: URL? { self.url(for: \.header) }
```
Comment (NachoSoto) [inline RevenueCatUI/Data/TemplateViewConfiguration+Images.swift:14]: `KeyPath` FTW

File: RevenueCatUI/Data/TestData.swift [hunk 2]
```diff
@@ -256,6 +255,11 @@ internal enum TestData {
     )
     static let paywallHeaderImageName = "9a17e0a7_1689854430..jpeg"
     static let paywallBackgroundImageName = "9a17e0a7_1689854342..jpg"
+    static let images: PaywallData.Configuration.Images = .init(
+        header: Self.paywallHeaderImageName,
+        background: Self.paywallBackgroundImageName,
+        icon: Self.paywallHeaderImageName
+    )
```
Comment (NachoSoto) [inline RevenueCatUI/Data/TestData.swift:262]: This is much better than the array solution. Thanks @guido732 @lburdock for the idea.

File: RevenueCatUI/Templates/MultiPackageTemplate.swift [hunk 3]
```diff
@@ -231,20 +231,6 @@ private extension MultiPackageTemplateContent {
...
-private extension TemplateViewConfiguration {
-
-    var backgroundURL: URL? {
-        return self.imageURLs.first
-    }
-
-    var iconURL: URL? {
-        guard self.imageURLs.count >= 2 else { return nil }
-        return self.imageURLs[1]
-    }
```
Comment (NachoSoto) [inline RevenueCatUI/Templates/MultiPackageTemplate.swift:244]: No more index hacks here.

File: Tests/RevenueCatUITests/Helpers/DataExtensions.swift [hunk 4]
```diff
@@ -36,23 +28,11 @@ extension Offering {
...
-            $0
-               .enumerated()
-               .map { index, _ in "image_\(index + 1).jpg" }
-        }
-    }
-
-    /// Useful for templates where we need images to be in opposite order.
-    /// For example: the background being the first image.
-    var withReversedImages: Self {
-        return self.mapImages { $0.reversed() }
```
Comment (NachoSoto) [inline Tests/RevenueCatUITests/Helpers/DataExtensions.swift:49]: This hack isn't needed anymore either.

**Review by joshdholtz — APPROVED** [review]

**Pull request merged**